lib/NormalModule.js

```diff
@@ -867,9 +867,17 @@ class NormalModule extends Module {
 				return callback(error);
 			}
 
+			const hasBinaryOverride =
+				/^asset\b/.test(this.type) &&
```

Comment:
Let's prefer `this.generatorOptions.binary` over `/^asset\b/.test(this.type)`, we can have more types in future and some of them can use the same feature